File: two_sum.py
def two_sum(numbers, target):
    seen = {}

    for i, num in enumerate(numbers):
        complement = target - num

        if complement in seen:
            return [seen[complement], i]

        seen[num] = i

    return []


numbers = list(map(int, input("Enter numbers: ").split()))
target = int(input("Enter target: "))

result = two_sum(numbers, target)

if result:
    print(f"Indices: {result}")
else:
    print("No pair found.")

File: two_sum.js
function twoSum(nums, target) {
    const seen = new Map();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];

        if (seen.has (complement)) {
            return [seen.get(complement), i];

        }

        seen.set(nums[i], i);

        }

        return [];
    }

    const nums = [2, 7, 11, 15];
    const target = 9;
    
    console.log("Numbers:", nums);
    console.log("Target:", target);
    console.log("Answer:", twoSum(nums,target));
